src/views/LeaderRequest.js

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+import queryString from 'query-string';
+import { Button } from '../components';
+import {
+  sendProtestLeaderRequest,
+  isProtestValid
+} from './SignUp/utils';
+
+const ProtestFormLabel = styled.label`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  margin-bottom: 10px;
+  font-weight: 600;
+`;
+
+const ProtestFormInput = styled.input`
+  width: 100%;
+  padding: 6px 12px;
+  margin-bottom: 0;
+  font-size: 16px;
+  border: 1px solid #d2d2d2;
+  -webkit-appearance: none;
+`;
+
+export default function LeaderRequest(props) {
+  const [phoneNumber, setPhoneNumber] = useState('');
+  const history = useHistory();
+
+  async function submitLeaderRequest(phoneNumber) {
+    const { protest: protestId } = queryString.parse(window.location.search);
```

Comment:
I guess it's not critical but you can make this a react-router param and use it like I did with the protestId in the protest page

Reply:
We use `query` and not param here. You suggest to swith to a param?